### ISSUE TITLE
[WIP][macOS] Issues introduced by macOS 10.14

### ISSUE THREAD

**Issue by hrubymar10** [wesnoth/wesnoth#3455]
# Checklist
- [x] Issue 1) - Touch using trackpad in help - (e5d42478422f5751b81d3a53cf63cecb59f726a1)
- [x] Issue 2) - Game starts with blank screen (https://github.com/hrubymar10/MacCompileStuff/commit/1d1d7fc5bf49e9d54275ce577c6da953cc99f2ec)
- [x] Issue 3) - Master branch doesn't work at all
- [x] Issue 4) - Blurry video in window mode - (5a89822caa93481f37a9d4b61f524d34f9932ed0)
- [x] Issue 5) - Bad resolution in fullscreen - (5a89822caa93481f37a9d4b61f524d34f9932ed0)
- [ ] Issue 6) - OpenGL deprecation (Long term issue)

# Description of issues introduced by macOS 10.14
## 1) Touch using trackpad in help
If you open unit info inside help, view is scrolled by moving mouse cursor by trackpad.

## 2) Game starts with blank screen
Game starts with blank screen. If you will click into the blank window, you can hear click sound that some dialog was openned (depends on if you clicked on GUI button or not). Screen persist black until you move or resize window.

Problem is reproducible only while building using macOS 10.14 SDK and running it on macOS 10.14.

It doesn't matter if you will build package using Xcode 9 or Xcode 10. It is caused by macOS 10.14 SDK

Same build which doesn't work on macOS 10.14 works without problems on macOS 10.13

<img width="1552" alt="image" src="https://user-images.githubusercontent.com/2218817/44002879-53d2c9b0-9e4a-11e8-9460-b19f57b8675b.png">

## 3) Master branch doesn't work at all
It has similar symptoms as issue 2) but in this case you can't get video working even with resizing or moving window.
<img width="1122" alt="image" src="https://user-images.githubusercontent.com/2218817/44005858-642d3388-9e7a-11e8-8333-f08555cdd02c.png">

## 4) Blurry video in window mode
![2018-08-11 at 14 33](https://user-images.githubusercontent.com/2218817/43991753-bb57f06a-9d73-11e8-9ed5-c4640a04b9c2.png)

## 5) Bad resolution in fullscreen
When I switch game to fullscreen display starts to be cropped. Click coordinates works as expected (means that when you click somewhere in bottom right side, you will got window with credits or game quits or preferences appears (depends on where exactly you click)
<img width="1440" alt="image" src="https://user-images.githubusercontent.com/2218817/43991690-968280bc-9d72-11e8-963d-c7022bee9279.png">

## 6) OpenGL deprecation (Long term issue)
**Long story short:** Apple deprecated OpenGL in favour of Metal what sucks because wesnoth 1.15 was planned to migrate to OpenGL.

**Long story:**
From macOS 10.14 Beta 6 Release notes:
```
OpenGL and OpenCL Deprecation
Deprecations
• The APIs in the OpenGL and OpenCL frameworks are deprecated and remain present for compatibility purposes. Transition to Metal if your app is using OpenGL or OpenCL.
```

So far I see these options:
- Write our Metal renderer
- Use something like [MoltenGL](https://moltengl.com/)
- Use something else for wesnoth 1.15 (eg. [Vulkan](https://www.khronos.org/vulkan/))

**Comment by CelticMinstrel** [comment]
Regarding the last item, I think we should just stick to the shift to OpenGL. Deprecated means it'll still work for the time being, and by the time it stops working I'm sure there'll be something like MoltenGL that we can use. (We probably can't use MoltenGL itself because it seems to be non-free.)

**Comment by jostephd** [comment]
Are 2) and 3) fixed now that master has been replaced?

**Comment by GregoryLundberg** [comment]
cf. https://bugzilla.libsdl.org/show_bug.cgi?id=4272#c24

There appears to be a patch available. Don't know what that means about timeframe for general availability on macOS or homebrew or whatever

**Comment by hrubymar10** [comment]
3) is fixed probably for `master`

I'll try the SDL patch asap, it sounds promising.

**Comment by hrubymar10** [comment]
Issue 2) resolved by new SDL2 version: https://github.com/hrubymar10/MacCompileStuff/commit/1d1d7fc5bf49e9d54275ce577c6da953cc99f2ec

**Comment by hrubymar10** [comment]
This issue was replaced by issue #4109
